ParabolInc/parabol
# chore(metrics): Consolidate GA4 `sign_up` events emission

## Description
- Moved mutation `loginSAML` to SDL pattern
- Consolidated fields needed for analytics inside `UserLoginPayload` into a new fragment `GA4Frag`
- Moved GA4 `sign_up` event emission into `handleSuccessfulLogin` method
- Fixed a couple of missing arguments for SAML login

## Demo
<img width="1506" alt="Screenshot 2023-03-16 at 14 47 32" src="https://user-images.githubusercontent.com/1879975/225637407-8ed52041-2985-49a0-b212-a1cf106761a0.png">


## Testing scenarios
- [ ] Sign up from Google
  - Sign up a new account from Google
  - Verify the `sign_up` event is emitted with `is_patient_0` property set

- [ ] Sign up using password
  - Sign up a new account user password
  - Verify the `sign_up` event is emitted with `is_patient_0` property set

- [ ] Sign up with SAML
  - Sign up a new account with SAML
  - Verify the `sign_up` event is emitted with `is_patient_0` property set

  - Login with Google, password or SAML
  - Verify no `sign_up` events are emitted

## Final checklist
